src/File.php

```diff
@@ -1503,4 +1503,10 @@ protected function getFilter()
     {
         return FileNameFilter::create();
     }
+
+
+    public function ToImage(): ?AssetContainer
```

Comment:
Strictly speaking, this method is not needed to introduce the ability to create variants with other extension. If we want to minimise the risk, we could ship the PR without it and still ship something useful.

So you could still do something like `$file->ToJPG()`. However, you couldn't perform image manipulation on the result.